### ISSUE TITLE
Unsupported HTTP method: 'HEAD'

### ISSUE THREAD

**Issue by pkuzz** [alexkasko/rest-handlers#1]
Hello,

Recently in log I've found following lines

```
java.lang.IllegalArgumentException: Unsupported HTTP method: 'HEAD'
    at com.alexkasko.rest.handlers.HandlersDispatcher.dispatch(HandlersDispatcher.java:64)
    at com.alexkasko.rest.handlers.HandlersDispatcherServlet.dispatch(HandlersDispatcherServlet.java:79)
    at com.alexkasko.rest.handlers.HandlersDispatcherServlet.doGet(HandlersDispatcherServlet.java:47)
    at javax.servlet.http.HttpServlet.doHead(HttpServlet.java:241)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:640)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:723)
```

The problem is that in tomcat's 6 servlet api `HttpServlet` redirects `HEAD` method handling to `GET` handler, so we are getting exception in following lines of `HandlersDispatcher`

```
            if("GET".equals(req.getMethod())) mapping = registryGet;
            else if("POST".equals(req.getMethod())) mapping = registryPost;
            else if("PUT".equals(req.getMethod())) mapping = registryPut;
                // cannot happen, methods are filtered in servlet
            else throw new IllegalArgumentException("Unsupported HTTP method: '" + req.getMethod() + "'");
```

**Comment by alexkasko** [comment]
Hi,

On 03/31/2015 10:37 AM, pkuzz wrote:

> Hello,
> 
> Recently in log I've found following lines
> 
> ```
> java.lang.IllegalArgumentException: Unsupported HTTP method: 'HEAD'
>   at com.alexkasko.rest.handlers.HandlersDispatcher.dispatch(HandlersDispatcher.java:64)
>   at com.alexkasko.rest.handlers.HandlersDispatcherServlet.dispatch(HandlersDispatcherServlet.java:79)
>   at com.alexkasko.rest.handlers.HandlersDispatcherServlet.doGet(HandlersDispatcherServlet.java:47)
>   at javax.servlet.http.HttpServlet.doHead(HttpServlet.java:241)
>   at javax.servlet.http.HttpServlet.service(HttpServlet.java:640)
>   at javax.servlet.http.HttpServlet.service(HttpServlet.java:723)
> ```
> 
> The problem is that in tomcat's 6 servlet api `HttpServlet` redirects `HEAD` method handling to `GET` handler, so we are getting exception in following lines of `HandlersDispatcher`
> 
> ```
>              if("GET".equals(req.getMethod())) mapping = registryGet;
>              else if("POST".equals(req.getMethod())) mapping = registryPost;
>              else if("PUT".equals(req.getMethod())) mapping = registryPut;
>                  // cannot happen, methods are filtered in servlet
>              else throw new IllegalArgumentException("Unsupported HTTP method: '" + req.getMethod() + "'");
> ```

So change the check in first line to "GET" || "HEAD" should be enough? I 
cannot test the changes right now, can add a small change or merge a 
pull request though.

> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/alexkasko/rest-handlers/issues/1

## 

-Alex

**Comment by pkuzz** [comment]
Yes,

```
            if("GET".equals(req.getMethod()) || "HEAD".equals(req.getMethod())) mapping = registryGet;
```

would be enough (as I understand HEAD is the GET with Content-Length=0 and empty response body, and this `GET || HEAD` gets this behaviour exactly)

Checked this locally, works fine. And I suppose that creating a fork for pull request is an overhead for on-line change (correct me if I wrong, please).

**Comment by alexkasko** [comment]
On 04/01/2015 07:48 AM, pkuzz wrote:

> Yes,
> 
> ```
>              if("GET".equals(req.getMethod()) || "HEAD".equals(req.getMethod())) mapping = registryGet;
> ```

Added this and released 1.2.1 version.

> would be enough (as I understand HEAD is the GET with Content-Length=0 and empty response body, and this `GET || HEAD` gets this behaviour exactly)
> 
> Checked this locally, works fine. And I suppose that creating a fork for pull request is an overhead for on-line change (correct me if I wrong, please).
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/alexkasko/rest-handlers/issues/1#issuecomment-88370917

## 

-Alex

**Comment by pkuzz** [comment]
Everithing is ok now, thanks!
